src/xtgeo/grid3d/_grid_import_xtgcpgeom.py

```diff
@@ -66,21 +114,22 @@ def handle_metadata(result, meta, ncol, nrow, nlay):
     result["coordsv"] = coordsv.reshape((nncol, nnrow, 6)).astype(np.float64)
     result["zcornsv"] = zcornsv.reshape((nncol, nnrow, nnlay, 4)).astype(np.float32)
     result["actnumsv"] = actnumsv.reshape((ncol, nrow, nlay)).astype(np.int32)
+
     if "subgrids" in req:
         result["subgrids"] = convert_subgrids(req["subgrids"])
 
 
 def import_xtgcpgeom(
-    mfile, mmap=False
-):  # pylint: disable=too-many-locals, too-many-statements
+    mfile: _XTGeoFile,
+    mmap: bool = False,
+) -> _RESULT:  # pylint: disable=too-many-locals, too-many-statements
     """Using pure python for experimental grid geometry import."""
     #
     offset = 36
-    with open(mfile.file, "rb") as fhandle:
-        buf = fhandle.read(offset)
+    with _read_filelike(mfile.file, size=offset) as f:
```

Comment:
Wondering if this context manager should actually just live in sys or directly in _XTGeoFile. `header = mfile.read(offset)` would be nice, or `with mfile.open() as f`. Unrelated to this PR though

Reply:
Yeah, i had a similar observation. But it felt a bit out of scope.